src/nvim/ops.c

```diff
@@ -2345,9 +2349,11 @@ static void op_yank_reg(oparg_T *oap, bool message, yankreg_T *reg, bool append)
   }
 
   reg->y_size = yanklines;
-  reg->y_type = yanktype;     /* set the yank register type */
+  assert(yanktype <= UCHAR_MAX);
```

Comment:
as yanktype is not really a numerical quantity but essentially an enum (one of MLINE, MCHAR, MBLOCK, MAUTO), this assert doesn't really make sense. I think you can just drop this one, but if an assert is added for motiontype/yanktype anywhere, it is better to tightly check that is is one of the 4 (or sometimes less) legal values.